lib/services/clusterconfig.go

```diff
@@ -20,9 +20,80 @@ import (
 	"github.com/gravitational/trace"
 
 	"github.com/gravitational/teleport/api/types"
+	apiutils "github.com/gravitational/teleport/api/utils"
 	"github.com/gravitational/teleport/lib/utils"
 )
 
+// ClusterConfigDerivedResources holds a set of the ClusterConfig-derived
+// resources following the reorganization of RFD 28.
+type ClusterConfigDerivedResources struct {
+	types.ClusterAuditConfig
+	types.ClusterNetworkingConfig
+	types.SessionRecordingConfig
+}
+
+// NewDerivedResourcesFromClusterConfig converts a legacy ClusterConfig to the new
+// configuration resources described in RFD 28.
+// DELETE IN 8.0.0
+func NewDerivedResourcesFromClusterConfig(cc types.ClusterConfig) (*ClusterConfigDerivedResources, error) {
+	ccV3, ok := cc.(*types.ClusterConfigV3)
+	if !ok {
+		return nil, trace.BadParameter("unexpected ClusterConfig type %T", cc)
+	}
+
+	var (
+		auditConfig types.ClusterAuditConfig
+		netConfig   types.ClusterNetworkingConfig
+		recConfig   types.SessionRecordingConfig
+		err         error
+	)
+	if ccV3.Spec.Audit != nil {
+		auditConfig, err = types.NewClusterAuditConfig(*ccV3.Spec.Audit)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	if ccV3.Spec.ClusterNetworkingConfigSpecV2 != nil {
+		netConfig, err = types.NewClusterNetworkingConfigFromConfigFile(*ccV3.Spec.ClusterNetworkingConfigSpecV2)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	if ccV3.Spec.LegacySessionRecordingConfigSpec != nil {
+		proxyChecksHostKeys, err := apiutils.ParseBool(ccV3.Spec.LegacySessionRecordingConfigSpec.ProxyChecksHostKeys)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		recConfigSpec := types.SessionRecordingConfigSpecV2{
+			Mode:                ccV3.Spec.LegacySessionRecordingConfigSpec.Mode,
+			ProxyChecksHostKeys: types.NewBoolOption(proxyChecksHostKeys),
+		}
+		recConfig, err = types.NewSessionRecordingConfigFromConfigFile(recConfigSpec)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
```

Comment:
I added helpers for each of these in https://github.com/gravitational/teleport/pull/7702. If this merges first, I'll update this code in my PR.

Reply:
Already merged, I'll be grateful if you remember to improve this with your helpers.